disasm-test/Main.hs

```diff
@@ -675,7 +680,24 @@ processBitCode pfx file = do
     Right m  -> do
       let m' = AST.fixupOpaquePtrs m
       postParseTests m'
-      parsed <- liftIO $ printToTempFile "ll" (show (ppLLVM (ppModule m')))
+      llvmVersion <- gets llvmVer
+      llvmAssembly <-
+        case vcVersioning llvmVersion ^? (_Right . major) of
+          Nothing -> do liftIO $ putStrLn ( "warning: unknown LLVM version ("
+                                            <> showVC llvmVersion <> "), assuming 3.5")
+                        return $ ppLLVM35 $ llvmPP m'
+          Just v ->
+            case v of
+              3 -> case vcVersioning llvmVersion ^? (_Right . minor) of
+                     Just 5 -> return $ ppLLVM35 $ llvmPP m'
+                     Just 6 -> return $ ppLLVM36 $ llvmPP m'
+                     Just 7 -> return $ ppLLVM37 $ llvmPP m'
+                     Just 8 -> return $ ppLLVM38 $ llvmPP m'
+                     o -> if maybe True (< 5) o
+                          then return $ ppLLVM35 $ llvmPP m'
+                          else return $ ppLLVM38 $ llvmPP m'
+              _ -> return $ ppLLVM38 $ llvmPP m'  -- TODO: LLVM38 is the last one currently defined
```

Comment:
This version-matching logic ought to be in its own function, it for no other reason than it adds a significant amount of clutter to `processBitCode`.

Reply:
I'm headed that direction, just trying to make sure I'm capturing all the use-case patterns before I consolidate this.